openprescribing/media/js/src/analyse-chart.js

```diff
@@ -530,5 +529,110 @@ var analyseChart = {
+      if (/^Unhandled selection:/.test(error)) {
+        // If we can't display a total for this particular selection just bail
+        // out and show nothing
+        return;
+      }
+      throw error;
+    }
+    var allMonths = this.globalOptions.allMonths;
+    var lastMonth = allMonths[allMonths.length - 1];
+    var twelveMonthsAgo = allMonths[allMonths.length - 12];
+    var financialYearStart = getFinancialYearStart(lastMonth);
+    var data = this.globalOptions.data.numeratorData;
+    var activeMonth = this.globalOptions.activeMonth;
+    var itemsMonthTotal = 0, itemsYearTotal = 0, itemsFinancialYearTotal = 0;
+    var costMonthTotal = 0, costYearTotal = 0, costFinancialYearTotal = 0;
+    var entry;
+    for (var i = 0; i < data.length; i++) {
+      entry = data[i];
+      if (selected && ! selected[entry.row_id]) continue;
+      if (entry.date === activeMonth) {
+        itemsMonthTotal += entry.items;
+        costMonthTotal += entry.actual_cost;
+      }
+      if (entry.date >= twelveMonthsAgo) {
+        itemsYearTotal += entry.items;
+        costYearTotal += entry.actual_cost;
+        if (entry.date >= financialYearStart) {
+          itemsFinancialYearTotal += entry.items;
+          costFinancialYearTotal += entry.actual_cost;
+        }
+      }
+    }
+    var el = this.el.summaryTotals;
+    el.find('.js-friendly-numerator').text(this.globalOptions.friendly.friendlyNumerator);
+    el.find('.js-orgs-description').text(getOrgsDescription(this.globalOptions.org, this.globalOptions.orgIds));
+    el.find('.js-selected-month').text(formatDate(activeMonth));
+    el.find('.js-financial-year-range').text(formatDateRange(financialYearStart, lastMonth));
+    el.find('.js-year-range').text(formatDateRange(twelveMonthsAgo, lastMonth));
+    el.find('.js-cost-month-total').text(formatNum(costMonthTotal));
+    el.find('.js-cost-year-total').text(formatNum(costYearTotal));
+    el.find('.js-cost-financial-year-total').text(formatNum(costFinancialYearTotal));
+    el.find('.js-items-month-total').text(formatNum(itemsMonthTotal));
+    el.find('.js-items-year-total').text(formatNum(itemsYearTotal));
+    el.find('.js-items-financial-year-total').text(formatNum(itemsFinancialYearTotal));
+    el.show();
+  }
+
 };
+
+function formatDate(dateStr) {
+  return Highcharts.dateFormat("%b '%y", new Date(dateStr));
+}
+
+function formatNum(n) {
+  return Highcharts.numberFormat(n, 0);
+}
+
+function getFinancialYearStart(dateStr) {
+  var date = new Date(dateStr);
+  // Months are zero-indexed so 3 is April
+  var financialYear = date.getMonth() >= 3 ? date.getFullYear() : date.getFullYear() - 1;
+  var yearStart = new Date(financialYear, 3, 1);
+  return Highcharts.dateFormat('%Y-%m-%d', yearStart);
+}
+
+function formatDateRange(fromDateStr, toDateStr) {
+  var fromDate = new Date(fromDateStr);
+  var toDate = new Date(toDateStr);
+  var formatStr = (fromDate.getFullYear() === toDate.getFullYear()) ? "%b" : "%b '%y";
```

Comment:
It would be nice to have the financial YTD during the month of April be *Apr 2018* (as opposed to  *Apr - Apr 2018*)

Reply:
Tiny but good! I have amended